CRM/Event/Form/Task/Register.php

```diff
@@ -115,11 +115,17 @@ public function postProcess(): void {
       $duplicateContacts = 0;
       foreach ($this->_contactIds as $k => $dupeCheckContactId) {
         // Eliminate contacts that have already been assigned to this event.
-        $dupeCheck = new CRM_Event_BAO_Participant();
-        $dupeCheck->contact_id = $dupeCheckContactId;
-        $dupeCheck->event_id = $event_id;
-        $dupeCheck->find(TRUE);
-        if (!empty($dupeCheck->id)) {
+        if (
+          CRM_Event_BAO_Participant::exists(
+            $dupeCheckContactId,
+            $event_id,
+            FALSE,
+            TRUE,
+            [],
+            [],
+            TRUE
```

Comment:
The old implementation did include `is_test` participants - this doesn't feel correct.

Reply:
Same here: this should be `FALSE`.